src/training/stylegan.py

```diff
@@ -0,0 +1,107 @@
+import torch
+from src.logger.simple import Logger
+from src.storage.simple import Storage
+from src.losses import ocr
+from torch import nn, optim
+from torch.utils.data import DataLoader
+from loguru import logger
+
+
+class StyleGanTrainer:
+    def __init__(self,
+                 model: nn.Module,
+                 style_embedder: nn.Module,
+                 content_embedder: nn.Module,
+                 optimizer: optim.Optimizer,
+                 scheduler: optim.lr_scheduler._LRScheduler,
+                 train_dataloader: DataLoader,
+                 val_dataloader: DataLoader,
+                 storage: Storage,
+                 logger: Logger,
+                 total_epochs: int,
+                 device: str,
+                 coef_ocr_loss: float,
+                 coef_perceptual_loss: float,
+                 perceptual_loss: nn.Module,
+                 ocr_loss:nn.Module):
+        
+        self.device = device
+        self.model = model
+        self.optimizer = optimizer
+        self.scheduler = scheduler
+        self.train_dataloader = train_dataloader
+        self.val_dataloader = val_dataloader
+        self.total_epochs = total_epochs
+        self.logger = logger
+        self.storage = storage
+        self.ocr_loss = ocr_loss.to(device)
+        self.perceptual_loss = perceptual_loss.to(device)
+        self.coef_ocr = coef_ocr_loss
+        self.coef_perceptual = coef_perceptual_loss
```

Comment:
We have `ComposeLoss` so we could pass it as one parameter

Reply:
From one side I agree, from the other -- it's more clear and we also have different objects to pass to these losses, so let's leave them separately, we won't suffer from 2 extra lines.

Reply:
True, I forgot that we calculate loss between different objects here, no objections here anymore